### ISSUE TITLE
[no-call-expression] False negative with the new control flow syntax

### ISSUE THREAD

**Issue by Res42** [angular-eslint/angular-eslint#1672]
**Description and reproduction of the issue**

```JSON
{
  "rules": {
    "@angular-eslint/template/no-call-expression": "error"
  }
}
```

`no-call-expression` should trigger when calling a function with the new control flow syntax, but it isn't.

```HTML
<!-- does not trigger, but it should -->
@if (shouldRender()) {
  <div>testing</div>
}

<!-- triggers correctly -->
<div *ngIf="shouldRender()">testing</div>
<!--        ~~~~~~~~~~~~~~ -->
```

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `17.1.1` |
| `@angular-eslint/template-parser`         | `17.1.1` |
| `@typescript-eslint/parser`               | `6.13.2` |
| `ESLint`                                  | `8.48.0` |
| `node`                                    | `20.10.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 17.0.7
Node: 20.10.0
Package Manager: npm 10.2.3
OS: win32 x64

Angular: 17.0.7
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, platform-server, router, ssr

Package                              Version
--------------------------------------------------------------
@angular-devkit/architect            0.1700.7
@angular-devkit/build-angular        17.0.7
@angular-devkit/core                 17.0.7
@angular-devkit/schematics           17.0.7
@angular/cdk                         17.0.4
@angular/material                    17.0.4
@angular/material-date-fns-adapter   17.0.4
@schematics/angular                  17.0.7
rxjs                                 7.8.1
typescript                           5.2.2
webpack                              5.89.0
zone.js                              0.14.2
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
Thanks @Res42 makes sense, would you like to submit a PR for this?
